storage/atsd/calculator.go

```diff
@@ -0,0 +1,94 @@
+// Copyright 2014 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package atsd
+
+import (
+	atsdNet "github.com/axibase/atsd-api-go/net"
+	info "github.com/google/cadvisor/info/v1"
+	"strconv"
+)
+
+const (
+	containerCpuUsageSystemPct = "cadvisor.cpu.usage.system%"
+	containerCpuUsageTotalPct  = "cadvisor.cpu.usage.total%"
+	containerCpuUsageUserPct   = "cadvisor.cpu.usage.user%"
+
+	containerCpuHostUsageSystemPct = "cadvisor.cpu.host.usage.system%"
+	containerCpuHostUsageTotalPct  = "cadvisor.cpu.host.usage.total%"
+	containerCpuHostUsageUserPct   = "cadvisor.cpu.host.usage.user%"
+
+	containerCpuUsagePerCpuPct = "cadvisor.cpu.usage.percpu%"
+)
+
+var oldStats = map[string]*info.ContainerStats{}
+
+// derived stats which cannot be calculated within atsd for now
+func CalculateDerivedSeriesCpuCommands(entity string, stats *info.ContainerStats) []*atsdNet.SeriesCommand {
+	if ostats, ok := oldStats[entity]; ok {
+		if ostats.Timestamp.Before(stats.Timestamp) {
+			time := uint64(stats.Timestamp.UnixNano() / 1e6)
+			deltaT := float64(stats.Timestamp.UnixNano() - ostats.Timestamp.UnixNano())
+			cpuCount := float64(len(stats.Cpu.Usage.PerCpu))
```

Comment:
I don't think this these need to be floats. Inline the float conversion if you really must.